src/Conversation.php

```diff
@@ -172,7 +173,7 @@ public function stop()
      */
     public function cancel()
     {
-        return $this->updateStatus('cancelled');
+        return ($this->updateStatus('cancelled') && $this->clear());
```

Comment:
in eply to https://github.com/MBoretto/php-telegram-bot/pull/9#discussion_r55926194


What is the difference between `stop()` and `cancel()`?
The way I understood it, is that a *stopped* conversation is one that's ended normally and a *cancelled* conversation has been cancelled somewhere in the middle of a conversation.
Is this right?

Since there's no way to "resume" a conversation at the moment, both `stop()` and `cancel()` stop and end the conversation, marking it as "finished" so to speak. So either one would need to be restarted as a new conversation afterwards.

Now about the separate variables in the class:
It could work both ways. Either as you say, or to modify the `update()` method to update the `$conversation` variable for us.

I still think it would be useful to have a specific getter and setter for the notes and a `revert()` command or something to undo any changes and not update them to the DB, but we'll discuss this in a later stage.